src/py_avro_schema/_alias.py

```diff
@@ -0,0 +1,70 @@
+"""
+Module to register aliases for Python types
+"""
+
+from collections import defaultdict
+
+FQN = str
+"""Fully qualified name for a Python type"""
+_ALIASES: dict[FQN, set[FQN]] = defaultdict(set)
+"""Maps the FQN of a Python type to a set of aliases"""
+
+
+def get_fully_qualified_name(py_type: type) -> str:
+    """Returns the fully qualified name for a Python type"""
+    module = getattr(py_type, "__module__", None)
+    qualname = getattr(py_type, "__qualname__", py_type.__name__)
+
+    # py-avro-schema does not consider <locals> in the namespace.
+    # we skip it here as well for consistency
+    if module and "<locals>" in qualname:
+        return f"{module}.{py_type.__name__}"
+
+    if module and module not in ("builtins", "__main__"):
+        return f"{module}.{qualname}"
+    return qualname
+
+
+def register_aliases(aliases: list[FQN]):
+    """
+    Decorator to register aliases for a given type.
+
+    Example::
+        @register_aliases(aliases=["py_avro_schema.OldAddress"])
+        class Address(TypedDict):
+            street: str
+            number: int
+    """
+
+    def _wrapper(cls):
+        """Wrapper function that updates the aliases dictionary"""
+        fqn = get_fully_qualified_name(cls)
+        _ALIASES[fqn].update(aliases)
+        return cls
+
+    return _wrapper
+
+
+def register_alias(alias: FQN):
+    """
+    Decorator to register a single alias for a given type.
+
+    Example::
+        @register_alias(alias="py_avro_schema.OldAddress")
+        class Address(TypedDict):
+            street: str
+            number: int
+    """
+
+    def _wrapper(cls):
+        """Wrapper function that updates the aliases dictionary"""
+        fqn = get_fully_qualified_name(cls)
+        _ALIASES[fqn].add(alias)
+        return cls
+
+    return _wrapper
+
+
+def get_aliases(fqn: str) -> list[str]:
+    """Returns the list of aliases for a given type"""
+    return list(_ALIASES.get(fqn) or [])
```

Comment:
question: should we order the returned list to be consistent in what we return in case there are multiple aliases? I don't know if it matters or not for the schema creation?

very nitpicky: to avoid the instantiation of the empty list every time, we could split this or something similar:

```suggestion
    if aliases := _ALIASES.get(fqn):
        return sorted(aliases)
    return []
```

Reply:
The order of the aliases should not matter. At least, I was not able to find anything about it mentioned in the docs.